### ISSUE TITLE
Exodus flatpak will not install or update on various distributions. 

### ISSUE THREAD

**Issue by storcke64** [flathub/io.exodus.Exodus#177]
Error: Error deploying: While trying to apply extra data: apply_extra script failed, exit status 256
error: Failed to install io.exodus.Exodus: Error deploying: While trying to apply extra data: apply_extra script failed, exit status 256

**Comment by develop7** [comment]
Confirming it, I'm experiencing same behavior.

**Comment by renich** [comment]
Same here on Fedora 38.

**Comment by renich** [comment]
```
[renich@introdesk ~]$ flatpak install --verbose io.exodus.Exodus
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/renich/.local/share/flatpak
Looking for matches…
F: Calling system helper: GenerateOciSummary
F: Fetching summary index file for remote ‘flathub’
F: Loading https://dl.flathub.org/repo/summary.idx using curl
F: Received 10001 bytes
F: Loaded indexed summary file d37d6c09f93d5f82c662566fb974a2ff94d33c3dd892ba0797f0f8aa05594605 from cache for remote ‘flathub’
F: Fetching summary index file for remote ‘flathub’
F: Loading https://dl.flathub.org/repo/summary.idx using curl
F: Received 10001 bytes
F: Loaded indexed summary file d37d6c09f93d5f82c662566fb974a2ff94d33c3dd892ba0797f0f8aa05594605 from cache for remote ‘flathub’
F: Transaction: install flathub:app/io.exodus.Exodus/x86_64/stable[*]
F: Looking for remote metadata updates for flathub
F: marking op install:app/io.exodus.Exodus/x86_64/stable resolved to 59e631d54207b67206842248cbde89002c0f45edb6e126cc065180f279857e85
F: Updating dependency org.freedesktop.Platform/x86_64/22.08 of io.exodus.Exodus/x86_64/stable
F: Transaction: update flathub:runtime/org.freedesktop.Platform/x86_64/22.08[$old]
F: marking op update:runtime/org.freedesktop.Platform/x86_64/22.08 resolved to 74b5ae6a53e3f39258f90c664a63ef256097bd77c4229fc11e96bc2dbd2ce0cf
F: Transaction: install/update flathub:runtime/org.freedesktop.Platform.GL.default/x86_64/22.08[*]
F: Transaction: install/update flathub:runtime/org.freedesktop.Platform.GL.default/x86_64/22.08-extra[*]
F: Transaction: install/update flathub:runtime/org.freedesktop.Platform.Locale/x86_64/22.08[/en]
F: Transaction: install/update flathub:runtime/org.gtk.Gtk3theme.Adwaita-dark/x86_64/3.22[*]
F: Transaction: install/update flathub:runtime/org.freedesktop.Platform.openh264/x86_64/2.2.0[*]
F: marking op install/update:runtime/org.freedesktop.Platform.openh264/x86_64/2.2.0 resolved to bf24f23f3ba385f6e8c9215ed94d979db99814b0b614504a23a6d0751dc5f063
F: marking op install/update:runtime/org.gtk.Gtk3theme.Adwaita-dark/x86_64/3.22 resolved to 7c055bbc0667524f34ca715abd67473480de93a176797a8fe4b7f7a71c4e3324
F: marking op install/update:runtime/org.freedesktop.Platform.Locale/x86_64/22.08 resolved to 4140d85aa7410bf434376fe37cd2b256d476d46e44ac028e781bce80be68b0a4
F: marking op install/update:runtime/org.freedesktop.Platform.GL.default/x86_64/22.08-extra resolved to e781cbdb0e17ea22365c55bcfdabfb6430c5f0a364c5307bdd5e1131c1a8680e
F: marking op install/update:runtime/org.freedesktop.Platform.GL.default/x86_64/22.08 resolved to 8b4ee41cce832ad87e6a806fd8359ac295ea23768cf515d88c502a0dcab54ca2
F: Checking installation ‘system’ for EOL unused refs
F: Checking installation ‘system’ by checking for dependent refs in ‘user’
F: Checking installation ‘system’ for EOL unused refs
F: Checking installation ‘system’ by checking for dependent refs in ‘user’

io.exodus.Exodus permissions:
    ipc	network	pulseaudio	x11	dri	tags [1]

    [1] proprietary


 1.	   	io.exodus.Exodus	stable	i	flathub	< 191.5 MB

Proceed with these changes to the system installation? [Y/n]: y
Installing…
F: Calling system helper: GetRevokefsFd
F: Calling system helper: GetRevokefsFd
F: flatpak_dir_pull: Using commit 59e631d54207b67206842248cbde89002c0f45edb6e126cc065180f279857e85 for pull of ref app/io.exodus.Exodus/x86_64/stable from remote flathub
Installing…                        0%  0 bytes/s
Installing… █                      5%
F: Loading https://downloads.exodus.com/releases/exodus-linux-x64-23.6.9.zip using curl
Installing… ██▌                   13%  11.7 MB/s
Installing… ███████               35%  28.5 MB/s
Installing… ██████████▊           54%  32.2 MB/s
Installing… ██████████████▊       74%  34.6 MB/s  00:01
Installing… ███████████████████   95%  36.2 MB/s  00:00
Installing… ████████████████████ 100%  38.1 MB/s  00:00
F: Received 190290799 bytes
F: Calling system helper: Deploy
Error: Error deploying: While trying to apply extra data: apply_extra script failed, exit status 256
error: Failed to install io.exodus.Exodus: Error deploying: While trying to apply extra data: apply_extra script failed, exit status 256
```

**Comment by renich** [comment]
Another one as root. Different error:

```
[root@introdesk ~]# flatpak install --verbose flathub io.exodus.Exodus
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /root/.local/share/flatpak
Looking for matches…
F: Fetching summary index file for remote ‘flathub’
F: Loading https://dl.flathub.org/repo/summary.idx using curl
F: Received 10001 bytes
F: Loaded indexed summary file d37d6c09f93d5f82c662566fb974a2ff94d33c3dd892ba0797f0f8aa05594605 from cache for remote ‘flathub’
F: Transaction: install flathub:app/io.exodus.Exodus/x86_64/stable[*]
F: Looking for remote metadata updates for flathub
F: marking op install:app/io.exodus.Exodus/x86_64/stable resolved to 59e631d54207b67206842248cbde89002c0f45edb6e126cc065180f279857e85
F: Updating dependency org.freedesktop.Platform/x86_64/22.08 of io.exodus.Exodus/x86_64/stable
F: Transaction: update flathub:runtime/org.freedesktop.Platform/x86_64/22.08[$old]
F: marking op update:runtime/org.freedesktop.Platform/x86_64/22.08 resolved to 74b5ae6a53e3f39258f90c664a63ef256097bd77c4229fc11e96bc2dbd2ce0cf
F: Transaction: install/update flathub:runtime/org.freedesktop.Platform.GL.default/x86_64/22.08[*]
F: Transaction: install/update flathub:runtime/org.freedesktop.Platform.GL.default/x86_64/22.08-extra[*]
F: Transaction: install/update flathub:runtime/org.freedesktop.Platform.Locale/x86_64/22.08[/en]
F: Transaction: install/update flathub:runtime/org.gtk.Gtk3theme.Adwaita-dark/x86_64/3.22[*]
F: Transaction: install/update flathub:runtime/org.freedesktop.Platform.openh264/x86_64/2.2.0[*]
F: marking op install/update:runtime/org.freedesktop.Platform.openh264/x86_64/2.2.0 resolved to bf24f23f3ba385f6e8c9215ed94d979db99814b0b614504a23a6d0751dc5f063
F: marking op install/update:runtime/org.gtk.Gtk3theme.Adwaita-dark/x86_64/3.22 resolved to 7c055bbc0667524f34ca715abd67473480de93a176797a8fe4b7f7a71c4e3324
F: marking op install/update:runtime/org.freedesktop.Platform.Locale/x86_64/22.08 resolved to 4140d85aa7410bf434376fe37cd2b256d476d46e44ac028e781bce80be68b0a4
F: marking op install/update:runtime/org.freedesktop.Platform.GL.default/x86_64/22.08-extra resolved to e781cbdb0e17ea22365c55bcfdabfb6430c5f0a364c5307bdd5e1131c1a8680e
F: marking op install/update:runtime/org.freedesktop.Platform.GL.default/x86_64/22.08 resolved to 8b4ee41cce832ad87e6a806fd8359ac295ea23768cf515d88c502a0dcab54ca2
F: Checking installation ‘system’ for EOL unused refs
F: Checking installation ‘system’ by checking for dependent refs in ‘user’
F: Checking installation ‘system’ for EOL unused refs
F: Checking installation ‘system’ by checking for dependent refs in ‘user’

io.exodus.Exodus permissions:
    ipc	network	pulseaudio	x11	dri	tags [1]

    [1] proprietary


 1.	   	io.exodus.Exodus	stable	i	flathub	< 191.5 MB

Proceed with these changes to the system installation? [Y/n]: y
Installing…
F: flatpak_dir_pull: Using commit 59e631d54207b67206842248cbde89002c0f45edb6e126cc065180f279857e85 for pull of ref app/io.exodus.Exodus/x86_64/stable from remote flathub
Installing…                        0%  0 bytes/s
Installing… █                      5%
F: Loading https://downloads.exodus.com/releases/exodus-linux-x64-23.6.9.zip using curl
Installing… ███▍                  17%  18.8 MB/s
Installing… ███████▌              38%  31.3 MB/s
Installing… ███████████▌          58%  34.6 MB/s
Installing… ███████████████▊      79%  37.1 MB/s  00:01
Installing… ████████████████████ 100%  38.1 MB/s  00:00
F: Received 190290799 bytes
F: No checksum specified, getting tip of app/io.exodus.Exodus/x86_64/stable from origin flathub
F: tip resolved to: 59e631d54207b67206842248cbde89002c0f45edb6e126cc065180f279857e85
F: extracting extra data to /var/lib/flatpak/app/io.exodus.Exodus/x86_64/stable/.59e631d54207b67206842248cbde89002c0f45edb6e126cc065180f279857e85-uuI6QH/files/extra
F: Opening system flatpak installation at path /var/lib/flatpak
F: /var/lib/flatpak/runtime/org.freedesktop.Platform/x86_64/22.08/74b5ae6a53e3f39258f90c664a63ef256097bd77c4229fc11e96bc2dbd2ce0cf/files/lib32 does not exist
F: Disallowing ipc access
F: Disallowing network access
F: Failed to run in transient scope: No systemd user session available, cgroups not available
F: Running /app/bin/apply_extra 
mv: cannot stat 'Exodus-linux-x64/*': No such file or directory
rm: cannot remove 'Exodus-linux-x64': No such file or directory
Error: While trying to apply extra data: apply_extra script failed, exit status 256
error: Failed to install io.exodus.Exodus: While trying to apply extra data: apply_extra script failed, exit status 256
```

**Comment by jnv** [comment]
Comparing the last working version (23.5.8) and the new one (23.6.9) makes the issue clear: previously the ZIP archive contained all files in `Exodus-linux-x64` which is what the script relied upon:

https://github.com/flathub/io.exodus.Exodus/blob/770c40775bbfc126d74659b07254a95e83a9026a/io.exodus.Exodus.json#L40

With the new version, the containing directory is missing.

So with very little understanding of Flatpak's build manifests, I think the build commands could be fixed like this:

```
[
  "unzip exodus.zip > /dev/null",
  "rm exodus.zip"
]
```

**Comment by jkoelker** [comment]
@jnv thanks for identifying the delta. I've submitted a PR that updates it. I've tested the build locally and it works as expected.

**Comment by jkoelker** [comment]
Looks like with https://github.com/flathub/io.exodus.Exodus/pull/179 the root folder is back.

**Comment by x80486** [comment]
@AHappyUser, would you be able to merge #179 any time soon?

**Comment by AHappyUser** [comment]
@x80486  Sorry, I have Been buried in work.  I just merged #179 now.

**Comment by renich** [comment]
Works fine in Fedora 38 x86_64. Thanks a lot, @AHappyUser.
